baseline-error-prone-annotations/src/main/java/com/palantir/errorprone/ForbidJavax.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.errorprone;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation applied to any method parameter which requires that all input types
+ * include at least one Jakarta EE 9 type, such as <code>jakarta.ws.rs.Path</code>
+ * or <code>jakarta.ws.rs.core.Feature</code> instead of the legacy <code>javax</code>
+ * versions of those.
+ *
+ * This sprinkling is required since many methods take a simple <code>Object</code> type
+ * and let Jersey determine what to do with the object at runtime. In order to obtain better
+ * compile time checking, this method can be applied.
+ */
+@Retention(RetentionPolicy.CLASS)
+@Target(ElementType.PARAMETER)
+public @interface ForbidJavax {}
```

Comment:
I think this would fit better in [conjure-java-runtime `conjure-java-annotations`](https://github.com/palantir/conjure-java-runtime/tree/develop/conjure-java-annotations/src/main/java/com/palantir/conjure/java/annotations), since we already define similar enforcement annotations there.

Reply:
So what I'm attempting by putting this here is to guarantee that this check lives on every classpath for "free", without having to push out a second check which adds the `errorprone` configuration to every build containing a separate errorprone project.

Is your suggestion simply to take this annotation and move it there, and then have baseline depend on that separate project, or move this entire check there?

Reply:
Yep, sorry I was unclear! The check would live in this repo and apply at ERROR, but the annotation would live in the other repo

Reply:
https://github.com/palantir/conjure-java-runtime/compare/feature/add-javax-errorprone-annotation?expand=1